x-pack/platform/plugins/shared/dataset_quality/public/hooks/use_dataset_quality_details_state.ts

```diff
@@ -160,8 +165,41 @@ export const useDatasetQualityDetailsState = () => {
     [service]
   );
 
+  const updateFailureStore = useCallback(
+    ({
+      failureStoreEnabled,
+      customRetentionPeriod,
+    }: {
+      failureStoreEnabled: boolean;
+      customRetentionPeriod?: string;
+    }) => {
+      service.send({
+        type: 'UPDATE_FAILURE_STORE',
+        failureStoreEnabled,
+        customRetentionPeriod,
+      });
+    },
+    [service]
+  );
+
+  // Refresh when failure store update completes successfully
+  useEffect(() => {
```

Comment:
We don't need this. We should handle the refresh entirely by sending the `UPDATE_FAILURE_STORE` action